File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.java
class Solution {
    public int minimumDifference(int[] nums, int k) {
      int mini = Integer.MAX_VALUE;
      Arrays.sort(nums);
      for(int i = nums.length - 1; i >= k - 1; i--){
        mini = Math.min(mini, nums[i] - nums[ i - k + 1]);
      }
    return mini;
    }
}

File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.cpp
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        sort(nums.begin(),nums.end());
        int diff =0;
        int ans =INT_MAX;
        int n = nums.size();

        if(k<=1)return 0;

        for(int i=0; i<=n-k; i++){
          diff= nums[i+k-1] - nums[i];
          ans= min(diff,ans);
        }

        return ans;
    }
};